enterprise/backup/src/main/java/org/neo4j/backup/BackupStageOutcome.java

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright (c) 2002-2017 "Neo Technology,"
+ * Network Engine for Objects in Lund AB [http://neotechnology.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Neo4j is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.neo4j.backup;
+
+public enum BackupStageOutcome
+{
+    SUCCESS,
+    WRONG_PROTOCOL,
+    TRUE_FAILURE,
+    ABSOLUTE_FAILURE;
```

Comment:
Better names and javadocs would be nice. `SUCCESS` makes sense, and `WRONG_PROTOCOL` you can infer is a failure condition, and you even get an idea about what the problem is. But `TRUE_FAILURE` and `ABSOLUTE_FAILURE` are not very helpful, though.

Reply:
kept `TRUE_FAILURE` as is, changed `ABSOLUTE_FAILURE` to `UNRECOVERABLE_FAILURE`, let me know if you have a better suggestion - the idea is that a true failure lets the surrounding class know that what it performed wasn't successful, but it can still try different methods (such as fallback from incremental to full, or fallback from full backup to another strategy). `UNRECOVERABLE_FAILURE` on the other hand means that there was a serious error and there cannot be any fallback.